analyze.js

```diff
@@ -104,19 +104,72 @@ async function attachScanFiles(scanGlobs, formData) {
+  if (typeof config.targetBranchName == 'undefined') {
+    core.info(`No target branch was defined. Using default target branch '${defaultBranch}'`)
+    config.targetBranchName = defaultBranch
+    config.baseBranchName = undefined
+  }
+
+  const baseBranchPresent = srmClient.checkIfBranchPresent(branches, config.baseBranchName)
+  const targetBranchPresent = srmClient.checkIfBranchPresent(branches, config.targetBranchName)
+
+  if (targetBranchPresent) {
+    core.info("Using existing SRM branch: " + config.targetBranchName);
+    // Not necessary, base branch is currently ignored in the backend if the target
+    // branch already exists. just setting to null to safeguard in case of future changes
+    config.baseBranchName = undefined
+  } else {
+    // Base branch is not defined; throw error
+    if (typeof config.baseBranchName == 'undefined') {
+      throw new Error("A parent branch must be specified when using a new target branch");
+    }
+
+    // Base branch is defined but is not present; cannot create target branch, so throw error
+    if (!baseBranchPresent) {
+      throw new Error("The specified parent branch does not exist: " + config.baseBranchName);
+    }
+
+    // If base branch is defined and present, create the target branch off the base
+    core.info(`Analysis will create a new branch named '${config.targetBranchName}' based on the branch '${config.baseBranchName}'`);
+  }
+}
+
 // most @actions toolkit packages have async methods
 module.exports = async function run() {
   const config = getConfig()
+  const MinimumVersionForBranching = '2022.4.3'
 
-  const client = new CodeDxApiClient(config.serverUrl, config.apiKey, config.caCert)
-  core.info("Checking connection to Code Dx...")
+  const client = new srmClient.SrmApiClient(config.serverUrl, config.apiKey, config.caCert)
+  core.info("Checking connection to SRM...")
 
-  const codedxVersion = await client.testConnection()
-  core.info("Confirmed - using Code Dx " + codedxVersion)
+  const srmVersion = await client.testConnection()
+  core.info("Confirmed - using SRM " + srmVersion)
 
   core.info("Checking API key permissions...")
   await client.validatePermissions(config.projectId)
-  core.info("Connection to Code Dx server is OK.")
+  core.info("Connection to SRM server is OK.")
+
+  core.info("Validating branch selection...")
+  // First check if the SRM version being used supports branching
+  if (srmVersion.localeCompare(MinimumVersionForBranching) < 0) {
+    core.info(
+        "The connected SRM server with version " + srmVersion + " does not support project branches. " +
+        "The minimum required version is " + MinimumVersionForBranching + ". The target branch and base " +
+        "branch options will be ignored."
+    )
+    // Set both branch configs to undefined since we will be ignoring them
+    config.baseBranchName = undefined
+    config.targetBranchName = undefined
```

Comment:
Assigning to `undefined` in JS is uncommon, standard practice is to set to `null` or use `delete object.field`. Using `undefined` explicitly is typically for cases where things like `null` and `undefined` are handled differently in some underlying system, but that's not the case here (well, it _is_ the case here since you check for `undefined` explicitly in various `if` clauses, but those should be changed to "falsey" checks anyway)

In this case I suggest setting to `null`, since `delete object.field` is typically only used if there's logic somewhere that is sensitive to the list of fields present in the object. With these changes we just need `baseBranchName` and `targetBranchName` to be "falsey", and setting to `null` is typical for representing "empty value"